csi/controller.go

```diff
@@ -1107,6 +1133,41 @@ func (s *OsdCsiServer) DeleteSnapshot(
 	return &csi.DeleteSnapshotResponse{}, nil
 }
 
+func (s *OsdCsiServer) deleteCloudBackup(
+	ctx context.Context,
+	req *csi.DeleteSnapshotRequest,
+) (*csi.DeleteSnapshotResponse, error) {
+	cloudBackupClient, err := s.getCloudBackupClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get any labels passed in by the CO
+	_, locator, _, err := s.specHandler.SpecFromOpts(req.GetSecrets())
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Unable to get parameters: %v", err)
+	}
+
+	credentialID := locator.VolumeLabels[osdSnapshotCredentialIDKey]
+
+	backupID := req.GetSnapshotId()
+
+	// Delete snapshot
+	resp, err := cloudBackupClient.Delete(ctx, &api.SdkCloudBackupDeleteRequest{
+		BackupId:     backupID,
+		CredentialId: credentialID,
+	})
+	if err != nil {
```

Comment:
Check for NotFound error here and return `return &csi.DeleteSnapshotResponse{}, nil` in case the underlying snapshot was deleted.

Reply:
Can we check similar to this line?
https://github.com/libopenstorage/openstorage/blob/master/csi/controller.go#L154

We want to check the actual status code

Reply:
String check is error prone, we want to check for actual status code.